### PR TITLE
Combined weight-part in cutting.

### PR DESCRIPTION
**Pull request by KaviyaPeriyasamy**
This PR contains code for handling combined-weight for the parts in cutting.
The creation of intermediate UOM, BOM, variants for the combined part.

### CODE REVIEW

File: apparelo/install.py [hunk 1]
```diff
@@ -48,3 +48,8 @@ def remove_defaults():
     stock_setting=frappe.get_doc("Stock Settings")
     stock_setting.stock_uom=None
     stock_setting.save()
+def create_new_uom():
```
Comment (karthikeyan5) [inline apparelo/install.py:54]: this function is not called in install.py anywhere?

**Review by karthikeyan5 — CHANGES_REQUESTED** [review]

File: apparelo/apparelo/doctype/cutting/cutting.py [hunk 2]
```diff
@@ -11,22 +11,29 @@
...
 		variants = []
 		if self.attribute_validate("Apparelo Colour", attribute_set["Apparelo Colour"]) and self.attribute_validate("Dia", attribute_set["Dia"]):
 			parts = set(self.get_attribute_values("Part"))
+			variant_attribute_set = {}
 			for part in parts:
-				variant_attribute_set = {}
-				variant_attribute_set['Part'] = [part]
+				variant_attribute_set['Part'] = []
+				variant_attribute_set['Part'].append(part)
+				if "," in part:
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/cutting/cutting.py]: can we use `is_combined` instead?

File: apparelo/apparelo/doctype/cutting/cutting.py [hunk 3]
```diff
@@ -11,22 +11,29 @@
...
+					for part_ in part.split(","):
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/cutting/cutting.py]: same here... can we use the list in the child table instead?

File: apparelo/apparelo/doctype/cutting/cutting.py [hunk 4]
```diff
@@ -59,39 +72,46 @@ def get_matching_details(self, part, size):
 
 	def create_boms(self, input_item_names, variants, attribute_set,item_size,colour,piece_count):
 		input_items = []
+		combo_=[]
+		boms = []
 		for input_item_name in input_item_names:
 			input_items.append(frappe.get_doc('Item', input_item_name))
-		boms = []
+		for part in attribute_set["Part"]:
+			if ',' in part:
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/cutting/cutting.py]: ditto

File: apparelo/apparelo/doctype/cutting/cutting.py [hunk 5]
```diff
@@ -59,39 +72,46 @@ def get_matching_details(self, part, size):
...
+				combined_part={}
+				variant=[]
+				variant_=[]
+				combined_part['color']=attribute_set["Apparelo Colour"]
+				combined_part['Size']=attribute_set["Apparelo Size"]
+				combined_part['Part']=part
+				count=part.count(",")
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/cutting/cutting.py]: ditto

File: apparelo/apparelo/doctype/cutting/cutting.py [hunk 6]
```diff
@@ -59,39 +72,46 @@ def get_matching_details(self, part, size):
...
-						else:
-							boms.append(existing_bom)
+			if combo_:
+				for combo in combo_:
+					if len(attr['Part'][0])<len(combo['Part']) and attr['Part'][0] in combo['Part']:
+						combo_bom=create_combined_bom(combo,variant)
+						boms.append(combo_bom)
+					else:
+						bom=create_common_bom(self,variant,attr,input_items)
+						if not ',' in bom:
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/cutting/cutting.py]: same here, can we not rely on comma being in the name?

File: apparelo/apparelo/doctype/cutting/cutting.py [hunk 7]
```diff
@@ -59,39 +72,46 @@ def get_matching_details(self, part, size):
...
+							boms.append(bom)
+			else:
+				bom_=create_common_bom(self,variant,attr,input_items)
+				if not ',' in bom_:
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/cutting/cutting.py]: ditto

**Review by karthikeyan5 — CHANGES_REQUESTED** [review]

File: apparelo/apparelo/doctype/item_production_detail/item_production_detail.py [hunk 8]
```diff
@@ -224,8 +224,9 @@ def create_process_details(self):
 								cutting_attribute=dict(counter_attr)
 								for value in cutting_attribute:
 									cutting_attribute[value]=list(set(cutting_attribute[value]))
-					process_variants['variants'] = list(set(variants_))
-					process_variants['BOM']=list(set(boms))
+					
+					process_variants['variants'] = list(set([var for var in variants_ if not ',' in var] ))
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/item_production_detail/item_production_detail.py]: same here

**Review by karthikeyan5 — CHANGES_REQUESTED** [review]

File: apparelo/apparelo/doctype/item_production_detail/item_production_detail.py (hunk 8 above)
Comment (KaviyaPeriyasamy) [reply apparelo/apparelo/doctype/item_production_detail/item_production_detail.py]: ## 800234b613a40f904c08949dcb001c68a8cc7f54
The above commit resolve all the changes requested

**Review by karthikeyan5 — DISMISSED** [review]

**Review by karthikeyan5 — APPROVED** [review]

**Pull request merged**